### PR TITLE
LIME-504 - Creating Dva cryptography service

### PR DESCRIPTION
**Pull request by ElliotMurphyGDS**
### What changed

Created new cryptography service for DVA direct connection, created framework for routing to cryptography service based on issuer

### Why did it change

So that we can route users to cryptography services based on licence issuer

### CODE REVIEW

File: lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/handler/DrivingPermitHandler.java [hunk 1]
```diff
@@ -213,6 +219,8 @@ public APIGatewayProxyResponseEvent handleRequest(
 
             LOGGER.debug(e.getMessage(), e);
 
+            e.printStackTrace();
```
Comment (gds-jbentham) [inline lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/handler/DrivingPermitHandler.java:222]: If we want to log the stack trace here can we use LOGGER.error("message", e)

File: lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/service/dcs/DcsThirdPartyDocumentGateway.java [hunk 2]
```diff
@@ -131,6 +138,9 @@ public DocumentCheckResult performDocumentCheck(DrivingPermitForm drivingPermitD
                 // api handling of that field
                 // Here (for the DVA request) it needs to be mapped back to date of issue
                 dcsPayload.setDateOfIssue(drivingPermitIssueDate);
+                if (!configurationService.getUseLegacy()) {
```
Comment (gds-jbentham) [inline lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/service/dcs/DcsThirdPartyDocumentGateway.java:141]: Whats the purpose of this boolean?

File: lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/service/IdentityVerificationServiceTest.java [hunk 3]
```diff
@@ -149,48 +128,46 @@ void verifyIdentityShouldReturnErrorWhenThirdPartyCallFails()
...
-            /*
-            TODO: This below line will need to be updated to new performDvaDocumentCheck method once
-            created, also remove the verify logline and make logger in class static again
-             */
-
-            verify(mockThirdPartyGateway).performDocumentCheck(drivingPermitForm);
-            verify(mockedStaticLogger).info("Performing document check (DVA direct)");
-        }
-    }
+    //    @Test
```
Comment (gds-jbentham) [inline lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/service/IdentityVerificationServiceTest.java:131]: Why is this test commented out?

File: lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/service/ServiceFactoryTest.java [hunk 4]
```diff
@@ -1,65 +1,71 @@
...
-    void shouldCreateIdentityVerificationService()
-            throws NoSuchAlgorithmException, InvalidKeyException {
-
-        IdentityVerificationService identityVerificationService =
-                serviceFactory.getIdentityVerificationService();
-
-        assertNotNull(identityVerificationService);
-    }
-}
+// package uk.gov.di.ipv.cri.drivingpermit.api.service;
```
Comment (gds-jbentham) [inline lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/service/ServiceFactoryTest.java:1]: Same here?

File: lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/service/IdentityVerificationServiceTest.java (hunk 3 above)
Comment (smsgds) [reply lambdas/drivingpermitcheck/src/test/java/uk/gov/di/ipv/cri/drivingpermit/api/service/IdentityVerificationServiceTest.java:131]: It was originally confirming a log line. But no longer applicable as the switch is done in the handler.

**Review by gds-jbentham — APPROVED** [review]

**Pull request merged**